pkg/runtask/v1beta1/kubernetes_test.go

```diff
@@ -0,0 +1,240 @@
+/*
+Copyright 2019 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/pkg/errors"
+
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	clientset "github.com/openebs/maya/pkg/client/generated/clientset/internalclientset"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func fakeGetClientsetOk() (cs *clientset.Clientset, err error) {
+	return &clientset.Clientset{}, nil
+}
+
+func fakeGetOk(cs *clientset.Clientset, name, namespace string,
+	opts metav1.GetOptions) (*apis.RunTask, error) {
+	return &apis.RunTask{}, nil
+}
+
+func fakeGetErr(cs *clientset.Clientset, name, namespace string,
+	opts metav1.GetOptions) (*apis.RunTask, error) {
+	return &apis.RunTask{}, errors.New("some error")
+}
+
+func fakeSetClientsetOk(k *Kubeclient) {
+	k.clientset = &clientset.Clientset{}
+}
+
+func fakeSetClientsetNil(k *Kubeclient) {
+	k.clientset = nil
+}
+
+func fakeGetClientsetNil() (clientset *clientset.Clientset, err error) {
+	return nil, nil
+}
+
+func fakeGetClientsetErr() (clientset *clientset.Clientset, err error) {
+	return nil, errors.New("Some error")
+}
+
+func fakeClientSet(k *Kubeclient) {}
+
+func TestWithDefaults(t *testing.T) {
+	tests := map[string]struct {
+		getFn              getFunc
+		getClientsetFn     getClientsetFunc
+		expectGet          bool
+		expectGetClientset bool
+	}{
+		// The current implementation of WithDefaults method can be
+		// tested using these two combinations only.
+		"When mockclient is empty": {nil, nil, false, false},
+		"When mockclient contains all of them": {fakeGetOk,
+			fakeGetClientsetOk, false, false},
+	}
+
+	for name, mock := range tests {
+		t.Run(name, func(t *testing.T) {
+			fc := &Kubeclient{
+				get:          mock.getFn,
+				getClientset: mock.getClientsetFn,
+			}
+
+			fc.withDefaults()
+			if mock.expectGet && fc.get == nil {
+				t.Fatalf(`test %s failed: expected non-nil fc.get
+but got %v`, name, fc.get)
+			}
+			if mock.expectGetClientset && fc.getClientset == nil {
+				t.Fatalf(`test %s failed: expected non-nil fc.getClientset
+but got %v`, name, fc.getClientset)
+			}
+		})
+	}
+}
+
+func TestWithClientset(t *testing.T) {
+	tests := map[string]struct {
+		clientset    *clientset.Clientset
+		isKubeClient bool
```

Comment:
can we rename this to `expectKubeClient`

Reply:
done.

Reply:
this is not reworked.. missed probably.

Reply:
It is in https://github.com/openebs/maya/pull/1057/commits/202b5a963bc930b339824378f5ab44dbbcf97953.

Reply:
done.